packages/app-desktop/SmalltalkfocusWrapper.ts

```diff
@@ -0,0 +1,206 @@
+			changeButtonFocus(dialog, names);
+		}
+		break;
+	}
+	event.stopPropagation();
+}
+
+function getDataName(el: HTMLElement): string {
+	return el.getAttribute('data-name')!.replace('js-', '');
+}
+
+function changeButtonFocus(dialog: HTMLElement, names: string[]): void {
+	const active = document.activeElement as HTMLElement;
+	const activeName = getDataName(active);
+	const isButton = /ok|cancel/.test(activeName);
+	const count = names.length - 1;
+	const getName = (activeName: string): string => (activeName === 'cancel' ? 'ok' : 'cancel');
+	if (activeName === 'input' || !count || !isButton) return;
+	const name = getName(activeName);
+	const elements = find(dialog, [name]);
+	for (const el of elements) {
+		el.focus();
+	}
+}
+
+const getIndex = (count: number, index: number): number => (index === count ? 0 : index + 1);
+
+function tab(dialog: HTMLElement, names: string[]): void {
+	const active = document.activeElement as HTMLElement;
+	const activeName = getDataName(active);
+	const count = names.length - 1;
+	const activeIndex = names.indexOf(activeName);
+	const index = getIndex(count, activeIndex);
+	const name = names[index];
+	const elements = find(dialog, [name]);
+	for (const el of elements) {
+		el.focus();
+	}
+}
+
+function closeDialog(el: HTMLElement, dialog: HTMLElement, ok: (result: string | null)=> any, cancel: ()=> any): void {
+	const name = el.getAttribute('data-name')!.replace('js-', '');
+	const keyCode = ('key' in event) ? (event as KeyboardEvent).key : String.fromCharCode((event as KeyboardEvent).keyCode);
+	if (/close|cancel/.test(name) || keyCode === 'Escape') {
+		// ESC case
+		cancel();
+		remove();
+		return;
+	}
+
+	let result: string | null = null;
+	if (name === 'ok') {
+		result = find(dialog, ['input']).reduce((prevValue: string | null, el: HTMLElement) => {
+			if (prevValue !== null) return prevValue;
+			if (el instanceof HTMLInputElement) return el.value;
+			return prevValue;
+		}, null);
+	}
+	ok(result);
+	remove();
+}
+
+function find(element: HTMLElement, names: string[]): HTMLElement[] {
+	const notEmpty = (a: any) => a;
+	const elements = names.map(name => element.querySelector(`[data-name="js-${name}"]`) as HTMLElement).filter(notEmpty);
+	return elements;
+}
+
+function addListenerAll(event: string, parent: HTMLElement, elements: string[], fn: (event: Event)=> void): void {
+	const foundElements = find(parent, elements);
+	for (const el of foundElements) {
+		el.addEventListener(event, fn);
+	}
+}
+
+function removeEl(name: string): void {
+	const el = document.querySelector(name) as HTMLElement;
+	el.parentElement!.removeChild(el);
+}
+
+function bind(fn: (...args: any[])=> any, ...args: any[]): ()=> any {
+	return () => fn(...args);
+}
+
+function createStore(value?: any) {
+	const data = {
+		value: value,
+	};
+
+	return function(newValue?: any) {
+		if (arguments.length === 0) return data.value;
+		data.value = newValue;
+		return newValue;
+	};
+}
```

Comment:
I don't understand what I'm looking at here. Is that the whole Smalltalk lib that's duplicated here?

Reply:
No, it's just the functions necessary for using the Smalltalk prompt function that I had to modify; the confirm function and the alert function, for example, are not in this wrapper because they didn't need modification to fix the bug.

Reply:
A wrapper is meant to wrap a library while here it looks like you're reimplementing the whole thing, and I'd rather not go for that kind of solution. I'm going to close the PR for now but feel free to submit another one